Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -384,7 +384,7 @@ ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels()
   mt->ParallelizeArray(
     0,
     allLabels.size(),
-    [this, &allLabels, &input, &labelsRegions, totalPixelCount, &labelsContoursOutput](SizeValueType i) {
+    [this, &allLabels, &input, &labelsRegions, totalPixelCount, &labelsContoursOutput](SizeValueType i) -> void {
```

Comment:
Is it necessary for a clean compilation to add `-> void` here? If so, can you please mention the warning that it has solved?

Reply:
https://open.cdash.org/viewBuildError.php?buildid=7093944 includes the following.  If I have misunderstood or if you see a better way, let's do that instead.
```plaintext
In file included from /usr/include4.8/memory:79:0,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkExceptionObject.h:22,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkMacro.h:1341,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkLightObject.h:21,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkObject.h:31,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkRegion.h:31,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkImageRegion.h:31,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkImage.h:21,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkConstNeighborhoodIterator.h:24,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkNeighborhoodIterator.h:24,
                 from /home/kitware/Dashboards/TestsModules/Core/Common/include/itkConstShapedNeighborhoodIterator.h:23,
                 from /home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:19:
/usr/include/c++/4.8/functional: In substitution of 'template<class _Res, class ... _ArgTypes> template<class _Functor> using _Invoke = decltype (std::__callable_functor(declval<_Functor&>())((declval<_ArgTypes>)()...)) [with _Functor = itk::ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels() [with TInputImage = itk::Image<unsigned char, 2u>]::__lambda8; _Res = void; _ArgTypes = {long unsigned int}]':
/usr/include4.8/functional:2257:9:   required from 'void itk::ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels() [with TInputImage = itk::Image<unsigned char, 2u>]'
/home/kitware/Dashboards/TestsModules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx:52:5:   required from 'void itk::ContourExtractor2DImageFilter<TInputImage>::GenerateData() [with TInputImage = itk::Image<unsigned char, 2u>]'
/home/kitware/Dashboards/TestsModules/Filtering/Path/test/itkContourExtractor2DImageFilterTest.cxx:758:1:   required from here
/usr/include4.8/functional:2181:71: error: use of 'itk::ContourExtractor2DImageFilter<TInputImage>::GenerateDataForLabels() [with TInputImage = itk::Image<unsigned char, 2u>]::__lambda8' before deduction of 'auto'
  using _Invoke = decltype(__callable_functor(std::declval<_Functor&>())
                                                                       ^
/usr/include4.8/functional:2257: confused by earlier errors, bailing out
Preprocessed source stored into /tmp/ccJDVoSG.out file, please attach this to your bugreport.
```

Reply:
@Leengit Thanks for these details! I'm sorry I don't really understand what's going on here. I did try to reproduce the gcc4.8 error at https://godbolt.org/z/6o5enc but it compiles fine without an explicit "-> void". 

Looking forward to say goodbye to GCC 4 🤔